src/Arcus.Messaging.Pumps.Abstractions/Resiliency/IMessagePumpCircuitBreaker.cs

```diff
@@ -23,12 +23,78 @@ public interface IMessagePumpCircuitBreaker
         /// <param name="configureOptions">The optional user-configurable options to manipulate the workings of the message pump interaction.</param>
         /// <exception cref="ArgumentException">Thrown when the <paramref name="jobId"/> is blank.</exception>
         Task PauseMessageProcessingAsync(string jobId, Action<MessagePumpCircuitBreakerOptions> configureOptions);
+    }
+
+    /// <summary>
+    /// Represents the available states in the <see cref="MessagePumpCircuitState"/> in which the message pump can transition into.
+    /// </summary>
+    internal enum CircuitBreakerState
+    {
+        Closed, HalfOpen, Open
+    }
+
+    /// <summary>
+    /// Represents the available states in which the <see cref="MessagePump"/> is presently in within the circuit breaker context
+    /// </summary>
+    public sealed class MessagePumpCircuitState
+    {
+        private readonly CircuitBreakerState _state;
+
+        private MessagePumpCircuitState(CircuitBreakerState state) : this(state, new MessagePumpCircuitBreakerOptions())
+        {
+        }
+
+        private MessagePumpCircuitState(CircuitBreakerState state, MessagePumpCircuitBreakerOptions options)
+        {
+            _state = state;
+
+            Options = options ?? throw new ArgumentNullException(nameof(options));
+        }
 
         /// <summary>
-        /// Continue the process of receiving messages in the message pump after a successful message handling.
+        /// Gets the boolean flag that indicates whether the message pump is in the 'Closed' circuit breaker state,
+        /// and should start retrieving messages.
         /// </summary>
-        /// <param name="jobId">The unique identifier to distinguish the message pump in the application services.</param>
-        /// <exception cref="ArgumentException">Thrown when the <paramref name="jobId"/> is blank.</exception>
-        Task ResumeMessageProcessingAsync(string jobId);
+        public bool IsClosed => _state is CircuitBreakerState.Closed;
+
+        /// <summary>
+        /// Gets the boolean flag that indicates whether the message pump is in the 'Open' circuit breaker state,
+        /// and should stop retrieving messages.
+        /// </summary>
+        public bool IsOpen => _state is CircuitBreakerState.Open;
+
+        /// <summary>
+        /// Gets the boolean flag tht indicates whether the message pump is in the 'Half-Open' circuit breaker state,
+        /// and should try retrieving a single message at a time before fully starting retrieving messages.
+        /// </summary>
```

Comment:
Minor remark, but if the pump is in a certain state, it is already working as defined by that state.
The comments on these properties say that 'the pump should ....'  , but 'the pump is ' actually already behaving in that way.
So, I'd change the wording of the comments in the XML comments to something like this :)

> Gets a boolean that indicates whether the message pump is in a 'Half-Open' state or not.

and then maybe in the remarks section some explanation:

> When the message pump is in the 'Half-Open' state, the pump retrieves a single message and verifies if the message can be processed.  If the message is processed succesfully, the pump will transition into the 'Closed' state.

Reply:
(And then some similar comments for the `IsClosed` and `IsOpen` properties :)

Reply:
Done in https://github.com/arcus-azure/arcus.messaging/pull/453/commits/bba3b25371bf6f05f45dd9179c57a2534d5c74c9